### PR TITLE
Added Tests

### PR DESCRIPTION
**Pull request by aman-chaintope**

### CODE REVIEW

File: backend/actions/address_detail.js [hunk 1]
```diff
@@ -133,4 +133,6 @@ app.get('/address/:address', (req, res) => {
     console.log(`Error retrieving information for addresss - ${urlAddress}. Error Message - ${err.message}`);
     logger.error(`Error retrieving information for addresss - ${urlAddress}. Error Message - ${err.message}`);
   });
-})
\ No newline at end of file
+})
+
+module.exports = {cl, elect};
```
Comment (azuchi) [inline backend/actions/address_detail.js:138]: Why export these?

File: backend/test/block.js [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+        });
+    });
+  
+    afterEach(function () {
+      sinon.restore();
+    });
+  });
+  
+  
+  describe("GET /block/:blockHash/rawData return type check", function() {
```
Comment (azuchi) [inline backend/test/block.js:87]:
```suggestion
  describe("GET /block/:blockHash/raw return type check", function() {
```

File: backend/test/transaction.js [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+          done();
+        });
+    });
+  
+    afterEach(function () {
+      sinon.restore();
+    });
+  });
+  
+  describe("GET /transaction/:txid/rawData return type check", function() {
```
Comment (azuchi) [inline backend/test/transaction.js:120]:
```suggestion
  describe("GET /transaction/:txid/raw return type check", function() {
```

File: backend/test/transaction.js [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+        });
+    });
+  
+    afterEach(function () {
+      sinon.restore();
+    });
+  });
+
+  
+  describe("GET /transaction/:txid/get return type check", function() {
```
Comment (azuchi) [inline backend/test/transaction.js:171]: Why do you need this API? Because similar API `/transaction/:txid` already exist.

**Review by azuchi — CHANGES_REQUESTED** [review]

File: backend/actions/address_detail.js (hunk 1 above)
Comment (aman-chaintope) [reply backend/actions/address_detail.js:138]: They are required for stubbing the instance while testing, Sir

File: backend/test/transaction.js (hunk 4 above)
Comment (aman-chaintope) [reply backend/test/transaction.js:171]: `/transaction/:txid` is used in the `/transaction` page on frontend and `/transaction/:txid/get` is used on `/address` page on frontend. Both call different RPCs on tapyrus-core.

Comment (azuchi) [reply backend/test/transaction.js:171]: What is it used for on the address page? This api looks call `gettransaction` RPC on Tapyrus Core, but that api is wallet api. explorer should not use wallet rpc in Tapyrus Core.

**Comment by rantan** [comment]
@aman-chaintope CI is failing. Can you check it?

File: backend/test/address.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
+/*const supertest = require("supertest");
```
Comment (rantan) [inline backend/test/address.js:1]: I guess this file will be added by Caching PR. So, I think these changes should not be added here.

**Comment by rantan** [comment]
I think the unit tests this PR added are reasonable :+1: 

But, I felt the file names in `/test` are difficult to understand correspondance between the test file and impl file.
How about using the name rule that is if the impl file name is `actions/block_list.js` then the test file should be `test/actions/block_list.spec.js`?

**Pull request merged**